src/screens/Books/index.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { t } from 'i18next';
+
+import BookDetail from '../../components/BookDetail';
+import { BookDetailContext } from '../../contexts/bookDetailContext';
+import { getBookDetail } from '../../services/bookService';
+import useRequest from '../../app/hooks/useRequest';
+
+function Books({ match }) {
+  const [isLoading, submitError, response] = useRequest({
+    request: getBookDetail,
+    payload: match.params.bookId
+  });
+
+  const bookDetail = response && response.ok ? response.data : {};
+  return (
+    <BookDetailContext.Provider value={bookDetail}>
```

Comment:
I wouldn't use Context here, I'd just do everything on the BookDetail component:

```
function BookDetail() {
  const [isLoading, submitError, response] = useRequest({
    request: getBookDetail,
    payload: match.params.bookId
  });

  return (
    <>
      {response?.ok && <p>{t('Errors:apiError')}</p>}}
      {isLoading && <p>{t('General:loadingContent')}...</p>}}
      {!isLoading && response.ok && (
        ...
      )}

    </>
  )
}
```